src/sentry/static/sentry/app/views/settings/incidentRules/ruleForm/index.tsx

```diff
@@ -105,6 +111,54 @@ class RuleFormContainer extends AsyncComponent<Props, State> {
     router.replace(recreateRoute('', {routes, params, location, stepBack: -2}));
   }
 
+  /**
+   * Checks to see if threshold is valid given target value, and state of
+   * inverted threshold as well as the *other* threshold
+   *
+   * @param type The threshold type to be updated
+   * @param value The new threshold value
+   */
+  isValidTrigger = (
+    triggerIndex: number,
+    trigger: Trigger,
+    errors,
+    changeObj?: Partial<Trigger>
+  ): boolean => {
+    const {alertThreshold, resolveThreshold} = trigger;
+
+    // If value and/or other value is empty
+    // then there are no checks to perform against
+    if (!hasThresholdValue(alertThreshold) || !hasThresholdValue(resolveThreshold)) {
+      return true;
+    }
+
+    // If this is alert threshold and not inverted, it can't be below resolve
+    // If this is alert threshold and inverted, it can't be above resolve
+    // If this is resolve threshold and not inverted, it can't be above resolve
+    // If this is resolve threshold and inverted, it can't be below resolve
+    const isValid =
+      trigger.thresholdType === AlertRuleThresholdType.BELOW
+        ? alertThreshold <= resolveThreshold
+        : alertThreshold >= resolveThreshold;
+
+    const otherErrors = errors.get(triggerIndex) || {};
+    const isResolveChanged = changeObj?.hasOwnProperty('resolveThreshold');
+    if (!isValid) {
+      errors.set(triggerIndex, {
+        ...otherErrors,
+        [isResolveChanged ? 'resolveThreshold' : 'alertThreshold']: isResolveChanged
+          ? trigger.thresholdType === AlertRuleThresholdType.BELOW
```

Comment:
I find nested ternary not so easy to read

Reply:
Updated